model/Delete/DeliveryDeleteService.php

```diff
@@ -104,6 +123,15 @@ public function getReport()
         return $this->report;
     }
 
+    /**
+     * @return array
+     */
+    protected function getWhiteListExceptions()
+    {
+        $exceptions = $this->getOption(static::OPTION_WHITE_LIST_EXCEPTIONS);
```

Comment:
I do not like configuring the service with white list exceptions. Any exception should be caught, otherwise task to delete delivery in the task queue fail. Currently we do not have ability to relaunch failed tasks.

Reply:
All the exceptions are caught. This list is for exceptions which are acceptable to not interrupt the process of deleting the delivery.
Example:
If you got common_NotFound exception and it's white listed the process of deleting the delivery it's not stopped, but if you got an \Exception the exception it's logged but the delivery it's not deleting entirely, because we don't want to delete a delivery with his executions not deleted properly.

Reply:
Why one exceptions are allowed and another one not? Why if `common_NotFound` exception was thrown we can proceed? We even do not know what exactly was not found.

Reply:
`common_exception_NotFound` this exception seems to be thrown when execution was trying to get his delivery  and the delivery was not there anymore, but because of this we don't want to interrupt deleting and failing of the task.

If you have other approach we can adjust, maybe removing the white list exception, and just log the error message, and not delete the delivery if one execution had failed, what do you think?